src/xml/attributes.cr

```diff
@@ -37,6 +37,23 @@ struct XML::Attributes
     find { |node| node.name == name }
   end
 
+  def delete(name : String)
+    curr = LibXML.xmlHasProp(@node, name)
+    if curr.null?
+      raise KeyError.new("Missing attribute: #{name}")
+    end
+    LibXML.xmlRemoveProp(curr)
+  end
+
+  def []=(name : String, value : String)
+    curr = self[name]?
+    if curr
+      curr.not_nil!.content = value
```

Comment:
This `not_nil!` is not needed because you already checked that it's not nil with `if curr` above